### PR TITLE
Added "silent_message" option for Telegram Agent

### PR DESCRIPTION
**Pull request by JohnnyKing94**
I opened a feature request (https://feathub.com/Tautulli/Tautulli/+444) this morning, then I tried to implement this on my own and it works. So I added a new checkbox in the notification telegram config in order to send new messages silently. The telegram users will receive a notification with no sound. In this way I can use the bot to spam non-important messages in the group (like watched notifications, play notification etc) without alerting everytime anyone.

### CODE REVIEW

**Review by agneevX — CHANGES_REQUESTED** [review]

File: plexpy/notifiers.py [hunk 1]
```diff
@@ -3460,6 +3464,12 @@ def _return_config_options(self):
                           'name': 'telegram_disable_web_preview',
                           'description': 'Disables automatic link previews for links in the message',
                           'input_type': 'checkbox'
+                          },
+                         {'label': 'Enable Silent Messages',
+                          'value': self.config['silent_message'],
+                          'name': 'telegram_silent_message',
+                          'description': 'Sends the message silently. Users will receive a notification with no sound.',
```
[the remark this answers is not in the document]
Comment (JonnyWong16) [reply plexpy/notifiers.py]: "`Send notifications silently without any alert sounds.`"

File: plexpy/notifiers.py [hunk 2]
```diff
@@ -3410,6 +3411,9 @@ def agent_notify(self, subject='', body='', action='', **kwargs):
 
         data['text'] = (text[:4093] + (text[4093:] and '...')).encode('utf-8')
 
+        if self.config['silent_message']:
```
Comment (JonnyWong16) [inline plexpy/notifiers.py]: Don't need to move this, just add the one line above.
```py
                if len(text) > 1024:
                    data['disable_notification'] = True
                else:
                    data['caption'] = text.encode('utf-8')
                    data['disable_notification'] = bool(self.config['silent_notification'])      
```

File: plexpy/notifiers.py [hunk 3]
```diff
@@ -3365,6 +3365,7 @@ class TELEGRAM(Notifier):
     _DEFAULT_CONFIG = {'bot_token': '',
                        'chat_id': '',
                        'disable_web_preview': 0,
+                       'silent_message': 0,
```
Comment (JonnyWong16) [inline plexpy/notifiers.py]: Rename the setting to `silent_notification`.

File: plexpy/notifiers.py [hunk 4]
```diff
@@ -3460,6 +3464,12 @@ def _return_config_options(self):
                           'name': 'telegram_disable_web_preview',
                           'description': 'Disables automatic link previews for links in the message',
                           'input_type': 'checkbox'
+                          },
+                         {'label': 'Enable Silent Messages',
```
[the remark this answers is not in the document]
Comment (JonnyWong16) [reply plexpy/notifiers.py]: "`Enable Silent Notifications`"

**Review by JonnyWong16 — CHANGES_REQUESTED** [review]

**Comment by agneevX** [comment]
Any progress on this @JohnnyKing94?

**Review by JohnnyKing94 — COMMENTED** [review]
Sorry for the long waiting... now everything has been adressed and tested... you can also try and let me know

**Comment by JonnyWong16** [comment]
Please also implement the wording changes I requested above.

**Comment by JohnnyKing94** [comment]
Done. :)

**Comment by JonnyWong16** [comment]
There is one more change. Rename `silent_message` to `silent_notification`.

**Comment by JohnnyKing94** [comment]
Done that too. 😄 😃

**Review by JonnyWong16 — APPROVED** [review]

**Comment by JonnyWong16** [comment]
Looks like everything works. Thanks.

**Pull request merged**